beacon_node/store/src/hot_cold_store.rs

```diff
@@ -977,40 +987,65 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold>
 
     /// Load a frozen state that lies between restore points.
     fn load_cold_intermediate_state(&self, slot: Slot) -> Result<BeaconState<E>, Error> {
+        if let Some(state) = self.state_cache.lock().get(&slot) {
+            return Ok(state.clone());
+        }
+
         // 1. Load the restore points either side of the intermediate state.
         let low_restore_point_idx = slot.as_u64() / self.config.slots_per_restore_point;
         let high_restore_point_idx = low_restore_point_idx + 1;
 
+        // Use low restore point as the base state.
+        let mut low_slot: Slot =
+            Slot::new(low_restore_point_idx * self.config.slots_per_restore_point);
+        let mut low_state: BeaconState<E> =
+            self.load_restore_point_by_index(low_restore_point_idx)?;
```

Comment:
If you can re-write this to avoid loading the low restore point until _after_ the cache is checked it will save ~200ms+. This function reads a whole BeaconState from disk, which is slow compared to loading a state from memory.